### ISSUE TITLE
make web-explorer "hash not found" page more user-friendly

### ISSUE THREAD

**Issue by GlenDC** [threefoldfoundation/tfchain#179]
The page should distinct between unlock hashes and other forms of hashes (identifiers).

Validate the characters of the hash first: ensure that all characters are hexadecimal characters (meaning one of the following: `0123456789ABCDEF`). If the hash contains an invalid character, inform the user **(1)**:

* that the hash is invalid, and what is invalid about it;
* explain what a valid hash can be;
* make some suggestions (e.g. is it a typo?, ...);

If all characters are valid, handle the rest of the logic differently based on the length:
 * length of 78: handle it as an unlock hash **(2)**;
 * length of 64: handle it as any other hash (coin/bs output ID, tx id, block ID) **(3)**;
 * any other length: inform the user that the hash is invalid, see **(1)**;

-+-
Handling of an unlock hash not found **(2)**:

Check the first 2 bytes:
* `00`: ensure the other 76 bytes are also `00`, if not, than that is probably a typo as the NilUnlockHash is always full zero, again inform the user about this;
* `01`, `02`, `03`: You can use it as a title, e.g.: Wallet Not Found. Once you did so, make sure to explain some reasons why this might be: the wallet didn't receive any coin or block stake outputs (spend or not), and nor is it referenced as part of a MultiSig Wallet;
* any other prefix: this is an unknown unlock hash type, and probably a typo: inform the user about this invalid unlock hash... etc, you know the drill;

-+-
Handling of a hash (identifier) not found **(3)**:

You can say something like "Hash Identifier Not Found" as the title. And continue with something along the lines of "The given identifier could be linked to any existing coin output, block stake output, transaction or block. Please check that the given hash is correct".

---

Don't make this dynamic not-found page too fancy, but please do layout it in a nice way.

Right now this page is way to cruel, just saying

```
Hash not Found in Database
Hash: 01f5e875005ec177d8c0d3ac7ebb1519196da7e2edfc8696007db1124accee4c6814f6f7fc1636
```

Does not help a user at all. And worse, it is not clear at all why this might be to an average user. For example the fact that a valid wallet address could return this in case it has no received outputs (spent or not), might not come to mind without us telling so.

**Comment by GlenDC** [comment]
@maximevanhees feel free to pick this one up as well.

**Comment by GlenDC** [comment]
Closed by a4f5f7268c8870351fab484742fa3f40ecf2cd7c and ab7f85ad1d213aabce9ce668f11ea62a100061e9.
